automation/src/test/scala/org/broadinstitute/dsde/firecloud/fixture/MethodFixtures.scala

```diff
@@ -1,49 +1,31 @@
 package org.broadinstitute.dsde.firecloud.fixture
 
 import org.broadinstitute.dsde.firecloud.config.AuthToken
-import org.broadinstitute.dsde.firecloud.fixture.MethodData.{SimpleMethod, SimpleMethodConfig}
 import org.broadinstitute.dsde.firecloud.test.{CleanUp, WebBrowserSpec}
 import org.broadinstitute.dsde.firecloud.util.Util.{appendUnderscore, makeUuid}
 import org.scalatest.Suite
 
 trait MethodFixtures extends CleanUp { self: WebBrowserSpec with Suite =>
 
 
-  def withConfigForRedactedMethodInWorkspace(testname:String, wsnamespace: String, wsname: String, withUnredactedSnapshots: Boolean)
-                                            (testCode: (String) => Any)
-                                            (implicit token: AuthToken): Unit = {
+  def withMethod(testName:String, method:Method, numSnapshots: Int = 1, cleanUp: Boolean = true)
+                (testCode: (String) => Any)
```

Comment:
it's actually easier to do the copy i think. that way the code generating the method doesn't have to do the copy and return a method.

Reply:
The code generating the method is here in one place so it would only have to be written once. There are currently multiple calls to `copy` that replace the method name, even multiple inside a single use of `withMethod`. Also, the fact that the test case's use of `SimpleMethod` (as opposed to some other method template) is repeated in the calls to `withMethod` and `copy`. It seems like a win to me.